### PR TITLE
feat(chat): redesign CLI invocation banner as collapsible structured chip

### PR DESCRIPTION
**Pull request by jamesbrink**
## What

Replaces the single-line horizontally-scrolling pre block at the top of every chat tab with a chip-style banner that summarizes the invocation when collapsed and expands to a column-aligned per-flag list on click.

## Before / After

**Before** — one long horizontally-scrolling line, no structure, no copy affordance, full binary path always visible:

```
/Users/jamesbrink/.local/bin/claude --print --output-format stream-json --input-format stream-json --verbose --include-partial-messages --permission-prompt-tool stdio --session-id 9cde1d6d-6c9b-48df-9765-5c8bcf522919 --model opus --mcp-conf…
```

**After (collapsed):**

```
▸ ⌨ claude · 12 flags                                              ⧉
```

**After (expanded):** binary name + faint full path on the first row, then a 2-column grid of flag names and values, with redacted values rendered in muted italic.

## Why

The previous banner read as raw terminal dump — visually noisy, no hierarchy, no copy button, and only the first ~120 chars were visible without horizontal scrolling. The new banner stays out of the way by default and reveals structure on demand.

## Implementation notes

- **Collapsed by default.** Per-session expand state lives in `sessionStorage` keyed by `sessionId` — it survives a refresh but doesn't bleed across tabs or bloat the Zustand store.
- **POSIX-aware tokenizer.** `tokenizeShellLine` honors single/double quotes and backslash escapes (matching what `src/agent/args.rs::shell_quote` emits). Splitting on whitespace alone would break paths-with-spaces.
- **Structured parser.** `parseInvocation` walks tokens to produce `{ binary, flags, prompt }`, recognizing `--flag=value`, `--flag value`, and boolean-flag forms.
- **Empty value cells.** Boolean flags emit an empty grid cell so the 2-column `display: contents` layout stays aligned across rows. (An earlier draft conditionally rendered the value cell, which let subsequent flag names slide into the value column — that's fixed.)
- **Redaction is unchanged.** `src/agent/args.rs::REDACTED_VALUE_FLAGS` (`--mcp-config`, `--settings`, `--append-system-prompt`, `--system-prompt`, `--agents`, `--betas`, `--json-schema`) already replaces sensitive values with `<redacted>` before persistence. The banner renders what's stored — no new redaction surface area.
- **Summary line is a placeholder.** `summarizeInvocation` currently returns `"${binary} · N flags"`. It's pinned by contract tests (must start with binary, never leak `<prompt>` or `<redacted>`, stay under ~80 chars) so future iterations can reshape the summary without breaking callers.

## i18n

5 new chat-namespace keys (`cli_invocation_expand`, `_collapse`, `_copy`, `_copied`, `_prompt_label`) translated across all five locales (en, es, ja, pt-BR, zh-CN).

## Tests

- 24 logic tests covering tokenizer edge cases (quoted strings, escapes, empty input), parser shapes (`--flag=value`, boolean flags, redacted values, the real screenshot invocation), `truncateMiddle`, and the `summarizeInvocation` contract.
- All 1473 existing tests still pass.
- Type check (`bunx tsc -b`), `bun run lint`, `bun run lint:css` clean.

## Out of scope

The companion `ClaudeFlagsSettings` / `ClaudeFlagRow` UI was prototyped in this branch but reverted before commit — the existing layout deserves a different rethink (semantic grouping or quick-toggle/all-flags split) rather than the card-row polish I tried first. Tracking separately.

### CODE REVIEW

File: src/ui/src/components/chat/cliInvocationBannerLogic.ts [hunk 1]
```diff
@@ -1,3 +1,208 @@
...
+      if (ch === "'") {
+        inSingle = false;
+        i += 1;
+        continue;
+      }
+      buf += ch;
+      hasContent = true;
+      i += 1;
+      continue;
+    }
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/cliInvocationBannerLogic.ts:74]: Addressed in c0b2e19e — good catch. The Rust backend (`shell_quote` in `src/agent/args.rs`) emits `'\\''` for embedded single quotes, which round-trips correctly only if `\\<char>` outside quotes is treated as a literal `<char>` (POSIX semantics). Added that branch to the unquoted-state of the tokenizer and added two round-trip tests covering both `O'Reilly` and the bare `\\<char>` escape (`cliInvocationBannerLogic.test.ts:50`).

File: src/ui/src/components/chat/CliInvocationBanner.tsx [hunk 2]
```diff
@@ -1,24 +1,165 @@
...
+    : null;
+
+  const [expanded, setExpanded] = useState<boolean>(() => {
+    if (!storageKey) return false;
+    try {
+      return sessionStorage.getItem(storageKey) === "1";
+    } catch {
+      return false;
+    }
+  });
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/CliInvocationBanner.tsx:46]: Addressed in c0b2e19e — accurate diagnosis. `useState`'s lazy initializer only runs on first mount, so the banner stayed mounted across tab switches with the previous tab's expand state. Added a `useEffect` that resyncs from `sessionStorage` on every `storageKey` change (`CliInvocationBanner.tsx:47`).

File: src/ui/src/components/chat/CliInvocationBanner.tsx [hunk 3]
```diff
@@ -1,24 +1,165 @@
...
+          className={styles.copyButton}
+          source={copySource}
+          tooltip={{
+            copy: t("cli_invocation_copy"),
+            copied: t("cli_invocation_copied"),
+          }}
+          ariaLabel={t("cli_invocation_copy")}
+          stopPropagation
+        />
+      </button>
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/CliInvocationBanner.tsx]: Addressed in c0b2e19e — restructured the header from a single wrapping `<button>` into a non-interactive `<div>` flex container with two sibling buttons: a wide left toggle (chevron + binary icon + summary) and the copy button on the right. Both buttons now have independent focus, ARIA, and click semantics, and the markup is valid HTML5.
